Settings.c

```diff
@@ -832,6 +836,14 @@ Settings* Settings_new(unsigned int initialCpuCount, Hashtable* dynamicMeters, H
    if (!realpath(this->initialFilename, this->filename))
       free_and_xStrdup(&this->filename, this->initialFilename);
 
+   /* check whether filename points to a regular file */
+   {
+      struct stat sb;
+      int err = lstat(this->filename, &sb);
+      if ((err == -1 && errno != ENOENT) || (err == 0 && !S_ISREG(sb.st_mode)))
```

Comment:
I try to align with the code style a few lines above.
```suggestion
      if (err && !S_ISREG(sb.st_mode) || (!err && errno != ENOENT))
```

Reply:
I think you mixed up the two OR operands.
Also we need to handle ENOENT differently here.

Reply:
Sorry. Probably this would work:
```suggestion
      if ((err && errno != ENOENT) || (!err && !S_ISREG(sb.st_mode)))
```